rclcpp/include/rclcpp/node_interfaces/node_base_interface.hpp

```diff
@@ -138,6 +138,10 @@ class NodeBaseInterface
   virtual
   std::unique_lock<std::recursive_mutex>
   acquire_notify_guard_condition_lock() const = 0;
+
+  RCLCPP_PUBLIC
+  virtual
+  ~NodeBaseInterface() = default;
```

Comment:
The destructor is conventionally at the top of the class just after any constructors.